lib/hanami/mailer.rb

```diff
@@ -223,8 +223,9 @@ def render(format)
     # @api private
     def deliver
       mail.deliver
-    rescue ArgumentError
-      raise MissingDeliveryDataError
+    rescue ArgumentError => e
+      raise MissingDeliveryDataError if e.message =~ /An SMTP (To|From) address/
+      raise
```

Comment:
WDYT maybe is better to drop guard style and use usual `if...else` for better reading here?